src/function/cast/vector_cast_helpers.cpp

```diff
@@ -2,6 +2,37 @@
 
 namespace duckdb {
 
+static bool isNull(const char *buf, idx_t start_pos, Vector &child, idx_t row_idx) {
+	if (buf[start_pos] == 'N' && buf[start_pos + 1] == 'U' && buf[start_pos + 2] == 'L' && buf[start_pos + 3] == 'L') {
```

Comment:
Is this always safe to do? Doesn't this function need to receive a `line_length` or similar to determine the end of the string?
Ah never mind, I read the code below where you check the length before calling this, and since it's a static function I guess it's fine to not have the safety built in to the function itself :)

Reply:
Also kind of an over-optimization probably but if the `&buf[start_pos]` is aligned to a 4-byte address, we can compare all 4 characters at the same time as a uint32_t (probably not worth the obfuscation this causes)

Reply:
ah yes my idea was to check the length beforehand to limit the amount of function calls.

Reply:
I believe the compiler will inline small functions like this wherever it can, but don't quote me on that haha